src/main/scala/cc/factorie/util/CircularBuffer.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of "FACTORIE" (Factor graphs, Imperative, Extensible)
+ * http://factorie.cs.umass.edu, http://github.com/factorie
+ * Licensed to the University of Massachusetts Amherst (UMass) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * UMass licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright C 2014, 2015, Oracle and/or its affiliates. All rights reserved.
+ */
+
+package cc.factorie.util
+
+import scala.reflect.ClassTag
+
+/**
+ */
+class CircularBuffer[A](val size : Int)(implicit m : ClassTag[A]) {
+  private val buffer = new Array[A](size)
+  private var position = 0
+
+  flush()
+
+  def flush() : Unit = {
+    var i = 0
+    while (i < size) {
+      buffer(i) = null.asInstanceOf[A]
+      i += 1
+    }
+  }
+
+  def add(item : A) : A = {
```

Comment:
So this is super nitpicky, but would it be too much trouble to implement this data structure more Scala-istically using += for add and apply for get? :)